### PR TITLE
TASK: NodeType mapping configurable | added more properties

### PR DESCRIPTION
**Pull request by ComiR**
Pretty much a complete overhaul.

See the [Readme](https://github.com/ElementareTeilchen/metadata-contentrepositoryadapter/blob/production/Readme.md) for new usage instructions.

This PR also includes some Bugfixes and adds many more properties to EXIF and IPTC NodeTypes.

Please also see https://github.com/neos/metadata/pull/4

### CODE REVIEW

**Comment by ComiR** [comment]
@daniellienert please review ;)

File: Classes/Mapper/ContentRepositoryMapper.php [hunk 1]
```diff
@@ -92,60 +87,52 @@ public function initializeObject()
     /**
      * @param Asset $asset
      * @param MetaDataCollection $metaDataCollection
-     * @throws NodeTypeNotFoundException
-     * @throws \TYPO3\Flow\Persistence\Exception\IllegalObjectTypeException
+     *
      * @return void
+     * @throws NodeTypeNotFoundException
```
Comment (daniellienert) [inline Classes/Mapper/ContentRepositoryMapper.php:92]: In Neos we don't annotate nested exception. So these annotations can be removed.

File: Classes/Mapper/ContentRepositoryMapper.php [hunk 2]
```diff
@@ -92,60 +87,52 @@ public function initializeObject()
...
     }
 
     /**
-     * @param AbstractNodeData $nodeData
+     * @param NodeTemplate $nodeData
      * @param NodeType $nodeType
      * @param MetaDataCollection $metaDataCollection
-     * @throws \TYPO3\Eel\Exception
+     *
+     * @throws EelException
```
Comment (daniellienert) [inline Classes/Mapper/ContentRepositoryMapper.php:119]: Nested Exception annotation can be removed.

File: Readme.md [hunk 3]
```diff
@@ -4,18 +4,61 @@ This package handles the mapping of meta data DTOs to the Neos Content Repositor
...
+    imageMetaDataNode = ${MetaData.find(q(node).property('image'), node)}
+    imageTitle = ${q(this.imageMetaDataNode).property('title')}
+}
+```
+
+### FlowQuery
+Get all meta data nodes matching the filter:
+
+```
+collection = ${q(metaDataRootNode).children('[instanceof Neos.MetaData:Exif][artist*="Daniel Lienert"]').get()}
```
Comment (daniellienert) [inline Readme.md]: The meta data root node is still `assets`, so the example should have assets here?

**Comment by daniellienert** [comment]
@ComiR: Looks great. Annotated some nitpicks. 
It would be good to have a new title for the feature, because the mentioned feature was dropped :)

**Pull request merged**